ledger/authority/src/bytes.rs

```diff
@@ -27,6 +27,18 @@ impl<N: Network> FromBytes for Authority<N> {
             2.. => Err(error("Invalid authority variant")),
         }
     }
+
+    /// Reads the authority from the buffer without performing any checks on the data.
+    fn read_le_unchecked<R: Read>(mut reader: R) -> IoResult<Self> {
+        // Read the variant.
+        let variant = u8::read_le(&mut reader)?;
+        // Match the variant.
+        match variant {
+            0 => Ok(Self::Beacon(FromBytes::read_le(&mut reader)?)),
```

Comment:
since `read_le_unchecked` defaults to `read_le`, shouldn't we always prefer it in case some underlying deserialization becomes unchecked in the future? perhaps this could also allow us to use a macro to create both impls by writing just one

Reply:
Made a relatively small set of changes in [this commit](https://github.com/ProvableHQ/snarkVM/pull/2789/commits/6771fbb019a272fc7600ecc85ab26763d660799d).